### PR TITLE
Add piwik tracking code

### PR DESCRIPTION
**Pull request by cbenz**
- [X] add tracker to single-page application router
- [X] add tracker to `/graph` page

### CODE REVIEW

**Comment by cbenz** [comment]
@MattiSG @fpagnoux It's difficult to test, because from `localhost` Piwik ignore HTTP requests.

But you can see with a [local production build](https://github.com/openfisca/legislation-explorer#production-server) that piwik HTTP requests are done, and not in development.

File: graph/index.html [hunk 1]
```diff
@@ -65,5 +65,21 @@
 <script src="//cdn.rawgit.com/bassjobsen/Bootstrap-3-Typeahead/master/bootstrap3-typeahead.min.js"></script>
 <script src="main.js"></script>
 <script>main();</script>
+<!-- Piwik -->
```
Comment (cbenz) [inline graph/index.html]: I anticipate: this whole file `graph/index.html` does not receive any environment configuration.

As a consequence it's not possible yet to execute this script conditionally.

But this is not an issue since Piwik ignores HTTP requests containing `localhost`.

We'll discuss at the next planning what to do with this file.

File: graph/index.html [hunk 2]
```diff
@@ -65,5 +65,21 @@
...
+  _paq.push(['enableLinkTracking']);
+  (function() {
+    var u="//stats.data.gouv.fr/";
+    _paq.push(['setTrackerUrl', u+'piwik.php']);
+    _paq.push(['setSiteId', '4']);
+    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
+    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'piwik.js'; s.parentNode.insertBefore(g,s);
+  })();
+</script>
+<noscript><p><img src="//stats.data.gouv.fr/piwik.php?idsite=4" style="border:0;" alt="" /></p></noscript>
```
Comment (MattiSG) [inline graph/index.html]: Please enforce `HTTPS` in images. Schema-less is deprecated, and we intend to only serve over HTTPS anyway.

File: graph/index.html [hunk 3]
```diff
@@ -65,5 +65,21 @@
...
 <script src="main.js"></script>
 <script>main();</script>
+<!-- Piwik -->
+<script type="text/javascript">
+  var _paq = _paq || [];
+  _paq.push(["setDomains", ["*.openfisca.fr","*.legislation.openfisca.fr","*.ui.openfisca.fr","*.www.openfisca.fr"]]);
+  _paq.push(['trackPageView']);
+  _paq.push(['enableLinkTracking']);
+  (function() {
+    var u="//stats.data.gouv.fr/";
```
Comment (MattiSG) [inline graph/index.html]: Please enforce `HTTPS`. Schema-less is deprecated, and we intend to only serve over HTTPS anyway.

File: graph/main.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-var initialVariableName = 'revdisp';
+var initialVariableName = 'revenu_disponible';
```
Comment (MattiSG) [inline graph/main.js]: This does not belong to this PR.

File: src/client.jsx [hunk 5]
```diff
@@ -14,11 +16,14 @@ addLocaleData(fr)
 export function renderApp() {
   const appMountNode = document.getElementById("app-mount-node")
   const initialState = window.__INITIAL_STATE__
+  const history = config.piwikConfig
```
Comment (MattiSG) [inline src/client.jsx]: Can it ever happen that `config.piwikConfig` is undefined?

File: config/production.france.js [hunk 6]
```diff
@@ -5,6 +5,10 @@ const HOST = process.env.HOST || "api.openfisca.fr",
   apiBaseUrl = process.env.API_URL || `https://${HOST}`,
   gitHubProject = "openfisca/openfisca-france",
   gitWebpageUrl = "https://github.com/openfisca/legislation-explorer",
+  piwikConfig = {
+    url: "stats.data.gouv.fr",
+    siteId: 4
```
Comment (MattiSG) [inline config/production.france.js]: I would love to have [`trackErrors`](https://github.com/joernroeder/piwik-react-router#trackerrors-false) too!

File: graph/index.html (hunk 3 above)
Comment (cbenz) [inline graph/index.html]: > Schema-less is deprecated

Interesting, I thought it was a good practice! Source: https://twitter.com/paul_irish/status/588502455530311680

Comment (MattiSG) [inline graph/index.html]: It did use to  ;)

File: config/production.france.js (hunk 6 above)
Comment (cbenz) [inline config/production.france.js]: I added it in 592ebb0 for the pages covered by the "react" part of the app.

The jquery/d3 part (/graph page) requires more work and I'm not sure it's worth doing it before we discuss the future of this page.

File: src/client.jsx (hunk 5 above)
Comment (cbenz) [inline src/client.jsx]: I changed the config files to export a `null` key explicitly if piwikConfig is not relevant.

File: config/production.tunisia.js [hunk 7]
```diff
@@ -5,6 +5,7 @@ const HOST = process.env.HOST || "api.openfisca.tn",
   apiBaseUrl = process.env.API_URL || `https://${HOST}`,
   gitHubProject = "openfisca/openfisca-tunisia",
   gitWebpageUrl = "https://github.com/openfisca/legislation-explorer",
+  piwikConfig = null,
```
Comment (MattiSG) [inline config/production.tunisia.js]: Why wouldn't we want to track Tunisia's page views? :wink:

Comment (cbenz) [inline config/production.tunisia.js]: Indeed, we can sort analytic logs by URL!

**Comment by MattiSG** [comment]
:shipit:

**Comment by cbenz** [comment]
Analytics can be seen here:
http://stats.data.gouv.fr/index.php?module=CoreHome&action=index&idSite=4&period=day&date=yesterday&segment=pageUrl%3D%40legislation.openfisca.fr#?module=Dashboard&action=embeddedIndex&idSite=4&period=day&date=yesterday&idDashboard=1&segment=pageUrl%3D@legislation.openfisca.fr
